src/View/mainpage/DrawROIWindow/Drawing.py

```diff
@@ -303,14 +407,15 @@ def mousePressEvent(self, event):
         ):
             self.drag_position = event.pos() - self.rect.topLeft()
         super().mousePressEvent(event)
-        is_coloured = (math.floor(
-            event.scenePos().x()), math.floor(event.scenePos().y())
-            ) in self.according_color_dict
+        x, y = linear_transform(
+            math.floor(event.scenePos().x()), math.floor(event.scenePos().y()),
+            self.rows, self.cols)
+        is_coloured = (x, y) in self.according_color_dict
         self.is_current_pixel_coloured = is_coloured
         self.draw_cursor(event.scenePos().x(), event.scenePos().y(),
                          self.draw_tool_radius, new_circle=True)
 
-        if self.is_current_pixel_coloured:
+        if not self.is_current_pixel_coloured:
```

Comment:
? Isn't this reversing the logic of the cursor in draw roi?? I don't think it's relevant to this PR

Reply:
Ah sorry, I switched it around when investigating and testing to add/remove points easier, i'll change it